app/lib/dns.server.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  Route53Client,
+  CreateHostedZoneCommand,
+  ChangeResourceRecordSetsCommand,
+  GetChangeCommand,
+} from "@aws-sdk/client-route-53";
+import {
+  CreateHostedZoneResponse,
+  ChangeResourceRecordSetsResponse,
+  GetChangeResponse,
+} from "@aws-sdk/client-route-53";
+import { isIPv4, isIPv6 } from "is-ip";
+
+export const route53Client = new Route53Client({
+  endpoint: process.env.AWS_ENDPOINT_URL || "http://localhost:5053",
+  region: process.env.AWS_REGION || "us-east-1",
+  credentials: {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID || "",
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY || "",
+    sessionToken: process.env.AWS_SESSION_TOKEN,
+  },
+});
+
+export const createHostedZone = async (domain: string) => {
+  try {
+    const command = new CreateHostedZoneCommand({
+      Name: domain,
+      CallerReference: new Date().toString(),
+    });
+    const response: CreateHostedZoneResponse = await route53Client.send(
+      command
+    );
+
+    if (!response.HostedZone?.Id) {
+      throw new Error("Missing hosted zone ID in AWS response");
+    }
+    return response.HostedZone?.Id.replace("/hostedzone/", "");
+  } catch (error) {
+    if (error instanceof Error) {
```

Comment:
Drop this if, get rid of `error.message` in line 41:

```ts
} catch(error) {
  // TODO: fix with better logging...
  console.warn(error);
  throw new Error(`Error while creating hosted zone`);
}
```

Please make this change in **all** code below.

Reply:
This hasn't been done yet.  You shouldn't need to check the type of error, since we won't use the `error.message`.

Reply:
You also need to do the same for all other methods in this file.